src/torrent/poll_kqueue.cc

```diff
@@ -279,7 +279,7 @@ PollKQueue::close(Event* event) {
       itr->udata = NULL;
 
   m_changedEvents = std::remove_if(m_changes, m_changes + m_changedEvents,
-                                   rak::equal(event, rak::mem_ref(&kevent::udata))) - m_changes;
+                                   [event](struct kevent& e) { return event == e.udata; }) - m_changes;
```

Comment:
should be using a const reference. struct is also optional in C++.

I wonder if adding a bool operatror== would allow changing remove_if to remove.

Reply:
or std::men_fn. Whichever is shorter.

Reply:
The new PR uses const, however it'd probably be better to avoid even simple logic changes in these hotfix-ish PRs.